### PR TITLE
Now build shows only errors and other logs are redirected

### PR DESCRIPTION
**Pull request by kmehant**
## Description
Travis build is being failed to huge logs from docker builds.
Now we redirect all normal logs and only show when there is an error during the build process.

### CODE REVIEW

**Comment by kmehant** [comment]
> Hey @kmehant,
> Something went wrong with the build.
> 
> TravisCI finished with status `errored`, which means the build failed because of something unrelated to the tests, such as a problem with a dependency or the build process itself.
> 
> [View build log](https://travis-ci.org/scorelab/Bassa/builds/688013746)
> 
> ###### TravisBuddy Request Identifier: 03bf9c90-9831-11ea-b97e-ed27cf305a90

@JaDogg @agentmilindu 
Everything is working fine now, but we need to set docker hub credentials in our Travis for Bassa so that we can push those images and the red cross is due to that only :)

**Comment by kmehant** [comment]
@agentmilindu @JaDogg 

Another choice is to have container builds.

And how about we manually pushing those images to the hub.

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+
+docker build --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-prod-ui ui >/dev/null
```
Comment (JaDogg) [inline deploy.sh:3]: Can you add some comments to this script. This is a bit hard to understand. 😁😁😁

Comment (kmehant) [reply deploy.sh:3]: Yupp :)

**Comment by kmehant** [comment]
@agentmilindu @JaDogg 
Please review 😁

**Comment by JaDogg** [comment]
@kmehant I've approved this. :)

**Comment by kmehant** [comment]
@agentmilindu I wish Milindu would give his review too

File: components/core/Dockerfile.prod [hunk 2]
```diff
@@ -19,7 +19,7 @@ RUN apt-get update && \
     apt-get clean
 
 # Installing the python modules
-RUN python3 setup.py develop
+RUN ["/bin/bash", "-c", "python3", "setup.py", "develop"]
```
Comment (agentmilindu) [inline components/core/Dockerfile.prod:22]: Just curious, why do we need this change? Isn't docker doing the same with Shell form?

File: .travis.yml [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-    docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD";
-    docker push scoreucsc/bassa-prod-ui;
-    docker push scoreucsc/bassa-dev-ui;
-    docker push scoreucsc/bassa-prod-server;
-    docker push scoreucsc/bassa-dev-server
-    docker push scoreucsc/bassa-aria2c;
-    fi
+script:
+  - docker build --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-prod-ui ui >/dev/null
+  - docker build -f ui/Dockerfile.dev --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-dev-ui ui >/dev/null
```
Comment (agentmilindu) [inline .travis.yml:14]: Do we have separate Docker repos as prod/dev? Can't we use docker tags to differentiate Dev and Prod images?

Comment (agentmilindu) [inline .travis.yml:14]: Can't we define the date at the beginning to an variable and then reuse it for all the commands? If we have to change it, then we have to change several places. And this does not look tidy.

Comment (agentmilindu) [reply .travis.yml:14]: Same goes with Git SHA

File: components/core/Dockerfile.prod (hunk 2 above)
Comment (kmehant) [reply components/core/Dockerfile.prod:22]: @agentmilindu 
True this is weird but I have to say this,

I have my docker run failing on my Mac if I do it with bash 
But it is other way round on Travis where it is failing due to `sh`.
What I feel is it's better to push it manually with sh itself. 
I am not sure why this is happening, do you think of any other workaround.

Also, when I approached a few dockerfile maintainers, and what I learned was they do the docker push part manually always.  
Please let me know your views?

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# shows only logs pushed to stderr
+docker build --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-prod-ui ui >/dev/null
+docker build -f ui/Dockerfile.dev --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-dev-ui ui >/dev/null
+docker build -f components/core/Dockerfile.prod --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-prod-server components/core >/dev/null
+docker build --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-dev-server components/core >/dev/null
+docker build --build-arg VCS_REF=`git rev-parse --short HEAD`   --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` -t scoreucsc/bassa-aria2c components/aria2c >/dev/null
+
+# push both production and development use docker images to dockerhub
+docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
+docker push scoreucsc/bassa-prod-ui
```
Comment (agentmilindu) [inline deploy.sh:14]: Think about tagging the Docker images with release version also. Push to both `latest` and release tags

File: .travis.yml (hunk 3 above)
Comment (kmehant) [reply .travis.yml:14]: That is cool will have something like 

these tags
`:prod` for production builds 
`:dev` for development builds 
is that okay?

Comment (kmehant) [reply .travis.yml:14]: Will do it. :)

File: deploy.sh (hunk 4 above)
Comment (kmehant) [reply deploy.sh:14]: Will do this too :)

File: components/core/Dockerfile.prod (hunk 2 above)
Comment (JaDogg) [reply components/core/Dockerfile.prod:22]: Add a comment and we can keep

**Comment by kmehant** [comment]
reopening this here https://github.com/scorelab/Bassa/pull/921
So closing this :)

**Pull request closed without merging**